test/test_ion_symbol.cpp

```diff
@@ -22,103 +22,93 @@
-    ION_ASSERT_OK(ion_writer_write_all_values(writer, reader)); \
-    ION_ASSERT_OK(ion_reader_close(reader)); \
-    cleanup_data; \
-    ION_ASSERT_OK(ion_test_writer_get_bytes(writer, stream, &result, &bytes_flushed)); \
-    if (as_binary) { \
-        assertBytesEqual(expected, expected_len, result, bytes_flushed); \
-    } \
-    else { \
-        assertStringsEqual(expected, (char *)result, bytes_flushed); \
-    } \
-    free(result);
+void populate_catalog(hCATALOG *pcatalog, ION_SYMBOL_TABLE *writer_imports[2]) {
+    hSYMTAB import1, import2;
+    ION_STRING import1_name, import2_name, sym1, sym2, sym3;
+    SID sid;
 
-#define _ION_SYMBOL_TEST_REWRITE_AND_ASSERT(retrieve_data, cleanup_data, as_binary, expected, expected_len) \
-    ION_WRITER_OPTIONS writer_options; \
-    ion_event_initialize_writer_options(&writer_options); \
-    ION_CATALOG *catalog = NULL; \
-    _ION_SYMBOL_TEST_REWRITE_AND_ASSERT_WITH_CATALOG(retrieve_data, cleanup_data, as_binary, expected, expected_len);
+    hCATALOG catalog = NULL;
 
-#define _ION_SYMBOL_TEST_REWRITE_FROM_BUFFER_AND_ASSERT_TEXT(base_macro, data, data_len, expected) \
-    ION_SYMBOL_TEST_DECLARE_WRITER; \
-    bytes_flushed = data_len; \
-    base_macro(result = data, /*do nothing*/, FALSE, expected, strlen(expected));
+    ION_ASSERT_OK(ion_string_from_cstr("import1", &import1_name));
+    ION_ASSERT_OK(ion_string_from_cstr("import2", &import2_name));
+    ION_ASSERT_OK(ion_string_from_cstr("sym1", &sym1));
+    ION_ASSERT_OK(ion_string_from_cstr("sym2", &sym2));
+    ION_ASSERT_OK(ion_string_from_cstr("sym3", &sym3));
+    ION_ASSERT_OK(ion_catalog_open(&catalog));
+    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[0], catalog, ist_SHARED));
+    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[1], catalog, ist_SHARED));
+    import1 = writer_imports[0];
+    import2 = writer_imports[1];
+    ION_ASSERT_OK(ion_symbol_table_set_name(import1, &import1_name));
+    ION_ASSERT_OK(ion_symbol_table_set_name(import2, &import2_name));
+    ION_ASSERT_OK(ion_symbol_table_set_version(import1, 1));
+    ION_ASSERT_OK(ion_symbol_table_set_version(import2, 1));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import1, &sym1, &sid));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym2, &sid));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym3, &sid));
+    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import1));
+    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import2));
+    *pcatalog = catalog;
+}
 
-#define ION_SYMBOL_TEST_REWRITE_WITH_CATALOG_FROM_BUFFER_AND_ASSERT_TEXT(data, data_len, expected) \
-    _ION_SYMBOL_TEST_REWRITE_FROM_BUFFER_AND_ASSERT_TEXT(_ION_SYMBOL_TEST_REWRITE_AND_ASSERT_WITH_CATALOG, data, data_len, expected);
+void open_writer_with_imports(bool is_binary, hWRITER *pwriter, ION_STREAM **pstream, ION_WRITER_OPTIONS *pwriter_options, ION_SYMBOL_TABLE **imports, SIZE import_count) {
```

Comment:
PR Tour 🛫: `open_writer_with_imports` replaces the `ION_SYMBOL_TEST_OPEN_WRITER_WITH_IMPORTS` macro, opening a new writer with the provided imports.